modopt/opt/algorithms.py

```diff
@@ -1236,7 +1236,7 @@ def get_notify_observers_kwargs(self):
         """
         return {
             'u_new': self._u_new,
-            'x_new': self._x_new,
+            'x_new': self._linear.adj_op(self._x_new),
```

Comment:
I think in order not to confuse anyone, we should rather have something like `x_primal` in addition to `x_new`. 

I generally think these observer things are quite bad because difficult to document but let's not make it worst here because it will be confusing for someone to create a metric and then the variable name is not the one corresponding to the algorithm's one.

Reply:
Well the change of name, in my opinion, must be done more generally throughout the codebase. 
If we change `x_new` to `x_primal`, a user cannot maintain the same mertric dictionary across POGM and FISTA or CondatVu. 
Do we still want to do it here? 
PS: This is still WIP, I may need to update a few codes for tests to pass

Reply:
I see what you mean, it's reasonable... although unfortunate.

Reply:
Yes, just that until now, we hadn't used metrics, I was about to write a method to comute SSIM for pysap-mri gridsearch, and just realized that this existed, wanted to revive it. 
Surely we can have a code cleanup based issue for it.